packages/opentelemetry-plugin-http/src/utils.ts

```diff
@@ -177,6 +177,66 @@ export const setSpanWithError = (
   span.setStatus(status);
 };
 
+/**
+ * Adds attributes for content-length and content-encoding HTTP headers
+ * @param { OutgoingHttpHeaders | IncomingHttpHeaders } headers http headers
+ * @param { Attributes } Attributes span attributes
+ */
+export const setRequestContentLengthAttribute = (
+  request: IncomingMessage,
+  attributes: Attributes
+) => {
+  const length = getContentLength(request.headers);
+  if (length === null) return;
+
+  if (isCompressed(request.headers)) {
+    attributes[HttpAttribute.HTTP_REQUEST_CONTENT_LENGTH] = length;
+  } else {
+    attributes[HttpAttribute.HTTP_REQUEST_CONTENT_LENGTH_UNCOMPRESSED] = length;
+  }
+};
+
+/**
+ * Adds attributes for content-length and content-encoding HTTP headers
+ * @param { OutgoingHttpHeaders | IncomingHttpHeaders } headers http headers
+ * @param { Attributes } Attributes span attributes
+ */
+export const setResponseContentLengthAttribute = (
+  response: IncomingMessage,
+  attributes: Attributes
+) => {
+  const length = getContentLength(response.headers);
+  if (length === null) return;
+
+  if (isCompressed(response.headers)) {
+    attributes[HttpAttribute.HTTP_RESPONSE_CONTENT_LENGTH] = length;
+  } else {
+    attributes[
+      HttpAttribute.HTTP_RESPONSE_CONTENT_LENGTH_UNCOMPRESSED
+    ] = length;
+  }
+};
+
+function getContentLength(
+  headers: OutgoingHttpHeaders | IncomingHttpHeaders
+): number | null {
+  const contentLengthHeader = headers['content-length'];
```

Comment:
This will fail if the casing is incorrect. Instead of using the headers object directly, I suggest you pass an `IncomingMessage | OutgoingMessage` which has a case-insensitive `getHeader(name)` function

Reply:
@dyladan Thanks for the feedback.  It seems like IncomingMessage does not have a `getHeader(name)` method.  
https://nodejs.org/api/http.html#http_class_http_incomingmessage
I think we can switch it up to passing in the response and request which does have `getHeader(name)`.

Reply:
Also, I commented before with a link, it seems all headers are automatically lower-cased. Also, it looks like other parts of code in this same file and this same package pull straight from the headers object.

Reply:
I think you're conflating the `IncomingMessage` class with the `IncomingMessage` type, which is just the request. But yes, that is what I was actually getting at.

Reply:
I'm not sure we need to be concerned about headers' cases or use the `getHeader()` method.

It seems that `getHeader()` is on `http.ClientRequest` and `http.ServerResponse` classes, but the function that we're tying into (`getIncomingRequestAttributes()`) has an `IncomingMessage` type coming in which doesn't have this method. We would have to re-write `getIncomingRequestAttributes()` and the functions which call it, to pass the `ClientRequest` object. That doesn't seem possible with the `_incomingRequestFunction()` function. It looks like we would have to change many layers of functions and their tests for us to use the `getHeader()` method.

But I also don't think we have to worry about headers' cases because the `headers` object on the `IncomingMessage` already has lowercased names according to the [docs](https://nodejs.org/api/http.html#http_message_headers). And looking at the code, it looks like it [lowercases all headers](https://github.com/nodejs/node/blob/0d474327a230f4934baf12322a18a831686100ec/lib/_http_incoming.js#L303).

Also, it seems like the code in this file and package already use this [headers object in this fashion](https://github.com/open-telemetry/opentelemetry-js/blob/v0.12.0/packages/opentelemetry-plugin-http/src/utils.ts#L388-L389).

Reply:
ok